Adds File Modal to view uploaded files

cedricwaxwing opened the pull request:
Adds parseCsvContent.tsx in lib/utils to handle csv parsing.
Creates FileModal.tsx to display file content.

Fixes #605

cedricwaxwing committed:
Adds File Modal and utils directory including parseCsvContent.tsx util

Fixes #605

cbrzn commented on apps/browser/components/Workspace.tsx:
```diff
@@ -27,6 +30,8 @@ function Workspace({ onUpload }: WorkspaceProps) {
 
   const workspaceLoading = workspaceUploads.length > 0;
 
+  const showFileModal = !!showFile;
```
i think we shouldn't depend on the file being loaded into state but rather have a state for opening the modal (and passing the file) - otherwise this happens (when u close the modal u can see the "View File"):

[Screencast from 20-12-23 23:54:51.webm](https://github.com/polywrap/evo.ninja/assets/21031138/00384b0d-cd60-4f8e-abe9-32968622f415)

cedricwaxwing replied:
Ah very good point. Fixed in https://github.com/polywrap/evo.ninja/pull/642/commits/171dded0eb74df4e85d9f3c4efdab8bb261be2f7

cedricwaxwing committed:
Separate concerns for modal open state / viewing file

Resolves https://github.com/polywrap/evo.ninja/pull/642#pullrequestreview-1791780622

dOrgJelli committed:
merge dev

dOrgJelli committed:
fix: support parsing csv's with multiple delimiter types
